src/request/view-models/LoggingViewModel.ts

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+/*tslint:disable:no-var-requires */
+const messageProcessor = require('../util/request-message-processor');
+/*tslint:enable:no-var-requires */
+
+import { ILogMessage } from '../messages/ILogMessage';
+import { IMessageEnvelope } from '../messages/IMessageEnvelope';
+
+import _ = require('lodash');
+
+export class LoggingViewModel {
+    private static getList = messageProcessor.getTypeMessageList;
+
+    private static options = {
+        'log-write': LoggingViewModel.getList
+    };
+
+    private _messages: ILogMessage[];
+
+    public get messages() {
+        return this._messages;
+    }
```

Comment:
The idea was that each instance of the `Logging` component get its own instance of the view model, which is initialized before component creation by the (same) request.  The message data held by the view model from that point should be effectively immutable (assuming my understanding of how/when React creates and renders components).  

Obviously things will change once filtering is involved, as those user actions will have notify the view model to sort/filter those messages in different ways, then the component will need to re-render, binding to the changes.